packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -373,6 +377,19 @@ export class ProjectLifecycleManager {
 
     this.ctx.update((s) => {
       s.currentProject = projectRoot
+      s.currentProjectGitInfo?.destroy()
+      s.currentProjectGitInfo = new GitDataSource({
+        isRunMode: this.ctx.isRunMode,
+        projectRoot,
+        onError: this.ctx.onError,
+        onBranchChange: () => {
+          this.ctx.emitter.branchChange()
+        },
+        onGitInfoChange: (specPaths) => {
+          this.ctx.emitter.gitInfoChange(specPaths)
+        },
```

Comment:
Can we do?

```suggestion
        onGitInfoChange: this.ctx.emitter.gitInfoChange,
```

Reply:
We'd need to add a bind or change `gitInfoChange (` to `gitInfoChange = ()` where it's defined.